### DISCUSSION TITLE
Should we define the different areas of the codebase and assign lead and co-maintainer?

### DISCUSSION THREAD

**Discussion by joschisan** [fedimint/fedimint#5470] [General]
## Motivation

* Track code ownership such that we now who to talk to if an issue or feature request arises
* Recognise areas that are deficient in maintenance and understanding
* More consistent code quality as mindshare is evenly distributed
* Consistently tie in efforts in a certain area over time
* Track code status
* No single point of failure, at least for critical parts of the codebase

## Proposal

Therefore I would propose to track the following information in the projects documentation, so it's clear for us but also for new contributors or developers from other projects who to talk to.

The people I assigned as maintainer that are not myself are obviously just a suggestion such as how to split the project into areas as well.

| Area | Lead-Maintainer | Co-Maintainers | Status |
| ------ | ------ | ------ | ------ |
| Core Server  | @joschisan | @elsirion | mostly well factored, no known issues | 
| Core Consensus  | @joschisan | @bradleystachurski | polished and documented | 
| Lightning Module | @joschisan | @m1sterc001guy | active development, known issues |
| Mint Module | @joschisan | @elsirion | active development, known issues |
| Wallet Module | @bradleystachurski  | @dpc @joschisan | active development, critical issues |
| Core Client | @dpc  | @elsirion @maan2003   | X |
| Lightning Gateway | @m1sterc001guy  | @joschisan  | X |
| Database | @m1sterc001guy  | @dpc | X |
| Client/Server API | @maan2003 | @oleonardolima | X |
| CI / Nix | @dpc  | @maan2003 @bradleystachurski @elsirion  | X |
| Testing | @bradleystachurski | X | X |
| Devimint | @maan2003  | X | X |
| Config Generation | X | X | X |

Personally I would appreciate a co-maintainer in my areas I lead which is why I had the idea for this. I would set up weekly calls in order to share ideas for prs, do deep dives etc. to set up context for pr review; I find it difficult to review prs from the code alone. @elsirion used the weekly calls when I was working on the consensus last year and it has produced really helpful reviews on my prs. The idea here is that leads and co-maintainer always have someone that is up to speed on their area and has a lot of context for reviewing prs, as the ideas behind where already discussed in the weeks leading up to the pr.

**Comment by dpc** [comment]
I can co-lead any of: devimint, testing, db, core client, mint module, core server.

BTW. "maintaince status" has a meh connotation of "obsolete"/"abandoned". I'm also unclear what's the difference between "complete" and "maintaince".

**Reply by joschisan** [reply]
I would define maintenance as mostly well factored and no known issues, however not as absolutely polished like consensus.

**Reply by joschisan** [reply]
Just changed it to this description
